Задача 2
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.
Для решения используем Z критерий
h0:m = m0
h1:m > m0

In [2]:
a = 0.05
n = 100
x = 17.5
m = 17
D = 4
std = D ** 0.5

In [3]:
z = (x - m)/(D ** 0.5 /(100 ** 0.5))
z

2.5

z табл = 1.645
z расчетное лежит правее z табличного и не попадает в область гипотезы
мы отвергаем гипотезу h0

Задача 3
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, 
что доверительная вероятность равна 99%? (Провести двусторонний тест.)
гипотеза:
h0: средние вес = 200
h1: средний вес != 200

In [4]:
import numpy as np
from scipy import stats

In [5]:
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [6]:
np.mean(x)

198.5

In [7]:
len(x)

10

In [8]:
np.std(x, ddof = 1)

4.453463071962462

a = 0.005
t01 = 3.25 и t02 = -3.25 табличная величина

In [9]:
t = (np.mean(x) - 200)/(np.std(x, ddof = 1)/len(x)**0.5)
t

-1.0651074037450896

In [ ]:
t наблюдаемое попадает в область гипотезы h0, гипотеза верна 

Задача 4
Задачу 4 решать с помощью функции. Есть ли статистически значимые различия в росте дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160
матери и дочери зависимые выборки
a = 0.05
h0: m = m0 рост дочерей зависит от роста матерей
h1: m != m0 рост дочерей не зависит от роста матерей

In [10]:
x = np.array([172, 177, 158, 170, 178,175, 164, 160, 169])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])

In [13]:
stats.ttest_rel(y, x)

Ttest_relResult(statistic=-0.559522990335608, pvalue=0.5911212354055175)

pvalue=0.5911212354055175 > a = 0.05 поэтому мы принимаем гипотезу h0

Задание 1
Когда используется критерий Стьюдента, а когда Z –критерий?
Z- критерий используют когда известна дисперсия генеральной совокупности и при большом размере выборки
Критерий Стьюдента используют если дисперсия не известна и выборка небольшьго размера